src/org/ods/usecase/JiraUseCaseZephyrSupport.groovy

```diff
@@ -4,24 +4,41 @@ import com.cloudbees.groovy.cps.NonCPS
 
 import org.ods.service.JiraZephyrService
 import org.ods.util.IPipelineSteps
+import org.ods.util.MROPipelineUtil
 import org.ods.util.SortUtil
 
 class JiraUseCaseZephyrSupport extends AbstractJiraUseCaseSupport {
 
     private JiraZephyrService zephyr
+    private MROPipelineUtil util
 
-    JiraUseCaseZephyrSupport(IPipelineSteps steps, JiraUseCase usecase, JiraZephyrService zephyr) {
+    JiraUseCaseZephyrSupport(IPipelineSteps steps, JiraUseCase usecase, JiraZephyrService zephyr, MROPipelineUtil util) {
         super(steps, usecase)
         this.zephyr = zephyr
+        this.util = util
     }
 
     void applyTestResultsAsTestExecutionStatii(List jiraTestIssues, Map testResults) {
         if (!this.usecase.jira) return
         if (!this.zephyr) return
 
+        def testCycleId = "-1"
+        if(!jiraTestIssues?.isEmpty()) {
+            def projectId = jiraTestIssues.first()?.projectId ?: ""
+            def versionId = this.getVersionId(projectId)
+            def cycleName = this.util.getBuildParams()?.targetEnvironmentToken
+            
+            def cycles = this.zephyr.getProjectCycles(projectId, versionId)
+            testCycleId = cycles.find { it.value instanceof Map && it.value.name == cycleName }?.key
```

Comment:
I would assume that you would also need to check for the version, no?

Reply:
@metmajer the call `def versionId = this.getVersionId(projectId)` would must return always a valid id: 
- -1 (in default case --> UNSCHEDULED) 
- or a real id if it find it.

Perhaps is the util object not properly initialized in vars/phaseInit.groovy?
`new JiraUseCaseZephyrSupport(steps, jiraUseCase, registry.get(JiraZephyrService.class.name), registry.get(PipelineUtil.class.name))`

Reply:
I'll have a check!